kong/plugins/moesif/base64.lua

```diff
@@ -86,51 +86,23 @@ function _M.encode( str, encoder, usecaching )
 	return concat( t )
 end
 
-function _M.decode( b64, decoder, usecaching )
-	decoder = decoder or DEFAULT_DECODER
-	local pattern = '[^%w%+%/%=]'
-	if decoder then
-		local s62, s63
-		for charcode, b64code in pairs( decoder ) do
-			if b64code == 62 then s62 = charcode
-			elseif b64code == 63 then s63 = charcode
-			end
-		end
-		pattern = ('[^%%w%%%s%%%s%%=]'):format( char(s62), char(s63) )
-	end
-	b64 = b64:gsub( pattern, '' )
-	local cache = usecaching and {}
-	local t, k = {}, 1
-	local n = #b64
-	local padding = b64:sub(-2) == '==' and 2 or b64:sub(-1) == '=' and 1 or 0
-	for i = 1, padding > 0 and n-4 or n, 4 do
-		local a, b, c, d = b64:byte( i, i+3 )
-		local s
-		if usecaching then
-			local v0 = a*0x1000000 + b*0x10000 + c*0x100 + d
-			s = cache[v0]
-			if not s then
-				local v = decoder[a]*0x40000 + decoder[b]*0x1000 + decoder[c]*0x40 + decoder[d]
-				s = char( extract(v,16,8), extract(v,8,8), extract(v,0,8))
-				cache[v0] = s
-			end
-		else
-			local v = decoder[a]*0x40000 + decoder[b]*0x1000 + decoder[c]*0x40 + decoder[d]
-			s = char( extract(v,16,8), extract(v,8,8), extract(v,0,8))
-		end
-		t[k] = s
-		k = k + 1
-	end
-	if padding == 1 then
-		local a, b, c = b64:byte( n-3, n-1 )
-		local v = decoder[a]*0x40000 + decoder[b]*0x1000 + decoder[c]*0x40
-		t[k] = char( extract(v,16,8), extract(v,8,8))
-	elseif padding == 2 then
-		local a, b = b64:byte( n-3, n-2 )
-		local v = decoder[a]*0x40000 + decoder[b]*0x1000
-		t[k] = char( extract(v,16,8))
-	end
-	return concat( t )
+
+-- character table string
+local b='ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/'
+
+function _M.decode(data)
```

Comment:
is it new updated base64.lua. Lets also put the source here in the comments for reference?

Reply:
i m suprised how its very concise now, not handling some cases as used to b4?

Reply:
does decode trim the data also before parsing?